cli/tests/test_split_command.rs

```diff
@@ -675,3 +680,199 @@ fn test_split_interactive_with_paths() {
+    std::fs::write(
+        test_env.set_up_fake_editor(),
+        ["", "next invocation\n", "write\nsecond-commit"].join("\0"),
+    )
+    .unwrap();
+    test_env.jj_cmd_ok(&main_path, &["split", "file2", "--parallel"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: vruxwmqv 97490ec0 second-commit
+    second: vruxwmqv 97490ec0 second-commit
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  wqnwkozpkust default@ second@ second-commit
+    │ ○  qpvuntsmwlqt first-commit
+    ├─╯
+    ◆  zzzzzzzzzzzz
+    "###);
+    let stderr = test_env.jj_cmd_failure(&secondary_path, &["log"]);
+    // The working copy for the second workspace will be stale.
+    insta::assert_snapshot!(stderr, @r###"
+    Error: The working copy is stale (not updated since operation ff670112b6f3).
+    Hint: Run `jj workspace update-stale` to update it.
+    See https://jj-vcs.github.io/jj/latest/working-copy/#stale-working-copy for more information.
+    "###);
+    test_env.jj_cmd_ok(&secondary_path, &["workspace", "update-stale"]);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  wqnwkozpkust default@ second@ second-commit
+    │ ○  qpvuntsmwlqt first-commit
+    ├─╯
+    ◆  zzzzzzzzzzzz
+    "###);
+}
+
+// A workspace should only have its working copy commit updated if the target
+// commit is the working copy commit.
+#[test]
+fn test_split_with_multiple_workspaces_different_working_copy() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "main"]);
+    let main_path = test_env.env_root().join("main");
+    let secondary_path = test_env.env_root().join("secondary");
+
+    test_env.jj_cmd_ok(&main_path, &["desc", "-m", "first-commit"]);
+    std::fs::write(main_path.join("file1"), "foo").unwrap();
+    std::fs::write(main_path.join("file2"), "foo").unwrap();
+
+    // Create the second workspace with a different working copy commit.
+    test_env.jj_cmd_ok(
+        &main_path,
+        &["workspace", "add", "--name", "second", "../secondary"],
+    );
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: qpvuntsm d12cb1c0 first-commit
+    second: pmmvwywv 44a7931a (empty) (no description set)
+    "###);
+    // Check the working-copy commit in each workspace in the log output. The "@"
+    // node in the graph indicates the current workspace's working-copy commit.
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  qpvuntsmwlqt default@ first-commit
+    │ ○  pmmvwywvzvvn second@
+    ├─╯
+    ◆  zzzzzzzzzzzz
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  pmmvwywvzvvn second@
+    │ ○  qpvuntsmwlqt default@ first-commit
+    ├─╯
+    ◆  zzzzzzzzzzzz
+    "###);
```

Comment:
nit: same comments as above apply here and below

Reply:
Ack